In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("SampleSuperstore.csv", encoding='latin-1')

In [3]:
print(f'Shape:{df.shape[0]:,}Rows *{df.shape[1]}Columns')

Shape:9,994Rows *13Columns


In [4]:
print(f'Memoryusage:{df.memory_usage(deep = True).sum()/1024**2:.1f}MB')

Memoryusage:4.9MB


In [5]:
print(f'Null_Values:{df.isnull().sum().sum()}Total')

Null_Values:0Total


In [6]:
print(f'Duplicates:{df.duplicated().sum()}Total')

Duplicates:17Total


In [7]:
print(f'Remove_Duplicates:{df.drop_duplicates(inplace= True)}Total')

Remove_Duplicates:NoneTotal


In [8]:
print(f'Duplicates:{df.duplicated().sum()}Total')

Duplicates:0Total


In [9]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 9977 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9977 non-null   object 
 1   Segment       9977 non-null   object 
 2   Country       9977 non-null   object 
 3   City          9977 non-null   object 
 4   State         9977 non-null   object 
 5   Postal Code   9977 non-null   int64  
 6   Region        9977 non-null   object 
 7   Category      9977 non-null   object 
 8   Sub-Category  9977 non-null   object 
 9   Sales         9977 non-null   float64
 10  Quantity      9977 non-null   int64  
 11  Discount      9977 non-null   float64
 12  Profit        9977 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1.1+ MB
None


In [10]:
print(df.head(0))

Empty DataFrame
Columns: [Ship Mode, Segment, Country, City, State, Postal Code, Region, Category, Sub-Category, Sales, Quantity, Discount, Profit]
Index: []


In [11]:
df.reset_index(drop=True, inplace=True)


In [12]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9977 entries, 0 to 9976
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9977 non-null   object 
 1   Segment       9977 non-null   object 
 2   Country       9977 non-null   object 
 3   City          9977 non-null   object 
 4   State         9977 non-null   object 
 5   Postal Code   9977 non-null   int64  
 6   Region        9977 non-null   object 
 7   Category      9977 non-null   object 
 8   Sub-Category  9977 non-null   object 
 9   Sales         9977 non-null   float64
 10  Quantity      9977 non-null   int64  
 11  Discount      9977 non-null   float64
 12  Profit        9977 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1013.4+ KB
None


In [14]:
df['Postal Code'] = df['Postal Code'].astype(str)

In [15]:
df.describe()

,Sales,Quantity,Discount,Profit
count,9977.000000,9977.000000,9977.000000,9977.00000
mean,230.148902,3.790719,0.156278,28.69013
std,623.721409,2.226657,0.206455,234.45784
min,0.444000,1.000000,0.000000,-6599.97800
25%,17.300000,2.000000,0.000000,1.72620
50%,54.816000,3.000000,0.200000,8.67100
75%,209.970000,5.000000,0.200000,29.37200
max,22638.480000,14.000000,0.800000,8399.97600


**I performed initial data quality checks using Pandas. The dataset contained 9,977 rows and 13 columns with no missing and earlier it has 17  duplicate values. I verified data types for numerical and categorical features and ensured the dataset was clean and ready for exploratory analysis and dashboarding.**

**Feature Engineering**

In [18]:
df['Profit_Margin'] = (df['Profit']/df['Sales']*100).round(2)

**Save_Cleaned_File**

In [19]:
print(f'Cleaned dataset saved to: {df.to_csv("Superstore_cleaned.csv",index = False)}')

Cleaned dataset saved to: None
